_posts/2023-01-12-websockets.md

```diff
@@ -0,0 +1,171 @@
+---
+layout: post
+title: "WebSockets: a comparison between open-source technologies"
+author: matteo.ricci
+categories: [ Software Development ]
+image: assets/images/websockets/0.png
+excerpt: "Benchmarking  WebSocket server performaces through a simple experiment"
+---
+
+## Introduction
+WebSocket is a communication protocol that supplies two-way communication between client and server.
+When a client starts a connection with a WebSockets server, this latter can push messages to all the connected clients. 
+That is in contrast with the **HTTP** client-server model, where a server cannot initiate a 
+message transfer to a client but can only respond to an **HTTP** request started by a client. 
+Further, by keeping the communication channel open, a WebSockets server allows the simultaneous sending and receiving 
+of messages among the connected clients through the server itself. 
+
+<center>
+    <img src="{{ site.url }}{{ site.baseurl }}/assets/images/websockets/1.png" width="400">
+    <br>
+    <em> A WebSocket server receives messages and notifies subscribers (Image generated by the author with E-Dell)</em>
+</center>
+
+In this work, we employed state-of-the-art, open-source, **JVM** technologies to assess the performances of a WebSocket server under progressive system overload. The focus of the study is comparing two event-loop models: **Spring Webflux** and **Ktor**.
+Nonetheless, some room for a didactical discussion will be left, explicitly showing the performance gap achieved by employing a thread-per-request Spring Boot WebSockets engine.
```

Comment:
You should write down the versions of the frameworks that you adopted for the experiment